modules/calypso-redirects.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+if ( ! class_exists( 'Jetpack_Calypso_Redirects' ) ) :
+
+class Jetpack_Calypso_Redirects {
+
+	public function __construct() {
+		if ( ! get_option( 'jetpack_calypso_redirects_active' ) || ! is_admin() ) {
+			return;
+		}
+
+		add_action( 'admin_menu', array( $this, 'menu_redirects' ), 30 );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+
+		add_filter( 'allowed_redirect_hosts', array( $this, 'allow_calypso_domain' ), 10, 1 );
+
+		add_action( 'load-toplevel_page_calypso_plugins', array( $this, 'plugins_redirect' ) );
+
+		if ( current_user_can( 'list_users' ) ) {
+			$this->register_user_redirect_handlers();
+		}
+	}
+
+	private function register_user_redirects() {
+		remove_submenu_page( 'users.php', 'users.php' );
+		remove_submenu_page( 'users.php', 'user-new.php' );
+		remove_submenu_page( 'users.php', 'profile.php' );
+
+		// replace each menu item one by one with its redirect
+		add_submenu_page( 'users.php', __( 'All Users' ), __( 'All Users' ), 'list_users', 'calypso_users', array( $this, 'users_redirect' ) );
```

Comment:
Always use `esc_html__` since it's safer than `__` and avoids rendering malicious HTML that might have been injected in a translation file.
Replace in all cases.

If you do need to use some HTML, you can use `wp_kses`.